### PR TITLE
git-town: use buildGoModule

### PR DESCRIPTION
**Pull request by aaronjheng**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by viraptor — APPROVED** [review]
Result of `nixpkgs-review pr 176200` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>git-town</li>
  </ul>
</details>

**Review by juboba — APPROVED** [review]

File: pkgs/tools/misc/git-town/default.nix [hunk 1]
```diff
@@ -1,23 +1,46 @@
...
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/git-town/git-town/src/cmd.version=v${version}"
+    "-X github.com/git-town/git-town/src/cmd.buildDate=nix"
+  ];
+
+  checkInputs = [ git ];
+  preCheck =
```
Comment (ajs124) [inline pkgs/tools/misc/git-town/default.nix:24]: can't you use `checkFlags` for this?

Comment (aaronjheng) [reply pkgs/tools/misc/git-town/default.nix:24]: Right now, `checkFlags` is not working. See https://github.com/NixOS/nixpkgs/pull/173702.

**Comment by superherointj** [comment]
Shell completion is missing. Please add it.

**Comment by aaronjheng** [comment]
> Shell completion is missing. Please add it.

Done.

**Comment by superherointj** [comment]
Please check if version is showing. (`./results/git-town/bin/git-town version`)

File: pkgs/tools/misc/git-town/default.nix [hunk 2]
```diff
@@ -1,23 +1,56 @@
...
+      ];
+    in
+    ''
+      HOME=$(mktemp -d)
+      # Disable tests requiring local operations
+      buildFlagsArray+=("-run" "[^(${builtins.concatStringsSep "|" skippedTests})]")
+    '';
+
+  postInstall = ''
+    # Install shell completions
```
Comment (superherointj) [inline pkgs/tools/misc/git-town/default.nix:43]:
```suggestion
```

**Comment by aaronjheng** [comment]
Done.

**Comment by superherointj** [comment]
Can you add a basic tests such as:
* simple execution (`git-town --help`)
* version ([see link](https://github.com/NixOS/nixpkgs/blob/master/doc/builders/testers.chapter.md#testversion-tester-testversion))

**Comment by aaronjheng** [comment]
Intended to. But it depends on `git`.
```
Error: cannot determine Git version: can't start subprocess 'git version': exec: "git": executable file not found in $PATH
```

**Comment by superherointj** [comment]
> Any suggestions?

Try something like:
```
  nativeBuildInputs = [ makeWrapper ];

    wrapProgram $out/bin/git-town \
        --prefix PATH : ${lib.makeBinPath [ git ]}
```

**Review by superherointj — APPROVED** [review]

**Pull request merged**